packages/zwave-js/src/lib/commandclass/ThermostatSetpointCC.ts

```diff
@@ -664,7 +664,9 @@ export class ThermostatSetpointCCReport extends ThermostatSetpointCC {
 			this.endpointIndex,
 			this._type,
 		);
-		valueDB.setValue(scaleValueId, this._scale.key);
+		if (!valueDB.hasValue(scaleValueId)) {
+			valueDB.setValue(scaleValueId, this._scale.key);
+		}
```

Comment:
This was specifically added for devices that can change the setpoint unit during usage:
https://github.com/zwave-js/node-zwave-js/pull/3693

only updating this when the value doesn't exit breaks that functionality.

Reply:
In that case, a compat flag disabling the scale learning would be more "correct", although this starts to go into territory I usually prefer to avoid - hyper specific flags tailored to individual devices. Lets do it though.

Reply:
Sounds good to me.